subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/AnnotationProcessorDetector.java

```diff
@@ -91,9 +83,11 @@ public void visitDependencies(TaskDependencyResolveContext context) {
         }, new MinimalFileSet() {
             @Override
             public Set<File> getFiles() {
+                // TODO:  This returns the entire compile classpath if any element of the
+                // compile classpath contains an annotation processor.  Why not just return
+                // the subset of elements that have annotation processors?
```

Comment:
I think we want to keep the full path for cases like defining a dependency that would not be required by the pom, for instance if you wanna override a dependency of an AP.